packages/docs/src/components/navigator.tsx

```diff
@@ -119,15 +132,29 @@ export class Navigator extends React.PureComponent<INavigatorProps, INavigatorSt
         });
     }
 
-    private getMatches() {
-        return filter(this.sections, this.state.query, {
-            key: "filterKey",
-        });
+    public componentDidUpdate() {
+        if (this.shouldCheckSelectedInViewport && this.menuRef != null) {
+            const selectedElement = this.menuRef.querySelector(`.${Classes.INTENT_PRIMARY}`) as HTMLElement;
+            const { offsetTop: selectedTop, offsetHeight: selectedHeight } = selectedElement;
```

Comment:
yes indeed i only learned of it recently! what do you think of this block? is it perhaps too convoluted, or is the renaming helpful?